### PR TITLE
Fix : registration not working after update

### PR DESCRIPTION
**Pull request by celiacheff**
Fix tchapgouv/tchap-web#186

Diff between this branch and 2.3.7 : https://github.com/tchapgouv/matrix-react-sdk/compare/v2.3.7-tchap...fix/registration-not-working

### CODE REVIEW

**Review by estellecomment — APPROVED** [review]

**Comment by estellecomment** [comment]
J'ai essayé de creer un compte externe invité, et une adresse agent, et j'ai eu dans les deux cas :

image

Screen Shot 2022-02-23 at 12 24 54

File: src/components/views/auth/RegistrationForm.js [hunk 1]
```diff
@@ -59,6 +60,7 @@ module.exports = React.createClass({
 
     getDefaultProps: function() {
         return {
+            minPasswordLength: 6,
```
Comment (estellecomment) [inline src/components/views/auth/RegistrationForm.js:63]: it says 8 in the help text

**Comment by estellecomment** [comment]
Missing translations : Maybe fixed by translations PR, if you don't fix in this PR, please keep track in the other so that we don't forget.

when using invalid email format : 
image

when email field is empty : 
image

Passwords don't match : 
image

Missing password : 
image

Password too short : 
image

Password too weak not sure if missing or not, because of my english-language browser ! 
image

**Comment by estellecomment** [comment]
My conclusion so far : 

 - there are two `register` flows running in parallel, one on `matrix.org` and one on tchap servers. The matrix.org one has a flow with captcha, the tchap one does not. I suspect that the first request of the two that returns makes the component change accordingly, and that is why we are not consistently seeing the captcha (sometimes it is there, sometimes not)
 
POST https://matrix.org/_matrix/client/r0/register
Response : {"session":"vpByfjfwNVBJZDWqspubKFZm","flows":[{"stages":["m.login.recaptcha","m.login.terms","m.login.email.identity"]}],"params":{"m.login.recaptcha":{"public_key":"6LcgI54UAAAAABGdGmruw6DdOocFpYVdjYBRe4zb"},"m.login.terms":{"policies":{"privacy_policy":{"version":"1.0","en":{"name":"Terms and Conditions","url":"https://matrix-client.matrix.org/_matrix/consent?v=1.0"}}}}}}

POST https://matrix.agent.dinum.tchap.gouv.fr/_matrix/client/r0/register
Response : {"session":"eDptzdCnKSncCELXnJlbLHVj","flows":[{"stages":["m.login.email.identity"]}],"params":{}}

**Comment by estellecomment** [comment]
In 2.3.7, the register request is also sent twice, but twice to tchap servers (instead of once to matrix.org and one to tchap server). 
It doesn't seem useful to send the request twice. So this was already a kind of bug, but invisible.

**Comment by celiacheff** [comment]
Didn't succeed to find why the first request is now matrix.org, but I fixed this by changing the hardcoded url. Here is my notes on the way: 

> web /src/vector/index.js: Matrix Chat init (mais pas de config sur HS/IS)
> MatrixChat.js: fallback to matrix.org -> getDefaultHsUrl l232
> MatrixChat.js: skipServerDetailsForRegistration mais je vois pas ou ça a un impact / utilisé dans Registration.js
> 
> Il faut obligatoirement un premier appel sur un HS (pas possible de juste faire un return si matrix.org)
> Pour minimiser les effets de bord, modification de l'url default codée en dur spécialement pour la registration (pourquoi c'est pas dans le config.json ???), plutôt que le config.json (qui marche aussi, mais side-effects à prévoir, à utiliser pour v4)

**Comment by estellecomment** [comment]
This is not great, and may not even work every time. But it's a start ! Merging.

**Pull request merged**